src/lsp/lp_doc.ml

```diff
@@ -12,6 +12,106 @@
 
 open Core
 module LSP = Lsp_base
+(*
+exception NoPosition of string *)
+
+module type RangeMap = sig
+
+  type point
+  type interval
+  type 'a t
+
+  val create_point : int -> int -> point
+  val l : point -> int
+  val c : point -> int
+
+  val interval_start : interval -> point
+  val interval_end : interval -> point
+
+  val create_interval : point -> point -> interval
+
+  type cmp = Before | In | After
+
+  val in_range : point -> interval -> cmp
+
+  val add : interval -> 'a -> 'a t -> 'a t
+  val find : point -> 'a t -> 'a option * interval
+
+  val empty : 'a t
+
+  val rangemap_to_string : ('a * string) t -> string
+end
+
+module M : RangeMap = struct
+
+  type point = { l : int; c : int }
+
+  type interval = point * point
+
+  let create_point l c = { l = l; c = c}
+
+  let l pt = pt.l
+
+  let c pt = pt .c
+
+  let interval_start (x, _) = x
+
+  let interval_end (_, y) = y
+
+  let create_interval x y =
+    assert ( (l x < l y) || (l x = l y && c x <= c y) );
+    (x, y)
+
+  type cmp = Before | In | After
+
+  let in_range pos (pos1, pos2) =
+
+    if pos.l < pos1.l || (pos.l = pos1.l && pos.c < pos1.c)
+    then Before
+
+    else if pos.l > pos2.l || (pos.l = pos2.l && pos.c > pos2.c)
+    then After
+
+    else In
+
+  type 'a t = (interval * 'a) list
+
+  let rec find (cursor_pos : point) (l: 'a t) = match l with
```

Comment:
coding style: `match` should come on next line. Please check coding style recommendations and see other files for examples.

Reply:
Solved in all new files by using an automatic formatter.